2006Redone Server/src/redone/game/shops/ShopAssistant.java

```diff
@@ -367,29 +369,27 @@ public boolean sellItem(int itemID, int fromSlot, int amount) {
 			}
 			// double ShopValue;
 			// double TotPrice;
-			int TotPrice2 = 0;
 			String itemName = ItemAssistant.getItemName(itemID).toLowerCase();
-			for (int i = amount; i > 0; i--) {
-				TotPrice2 = (int) Math.floor(getItemShopValue(itemID, 1, true));
-				if (player.getItemAssistant().freeSlots() > 0 || player.getItemAssistant().playerHasItem(995)) {
-					if (ItemDefinitions.getDef()[itemID].isNoteable == false) {
-						player.getItemAssistant().deleteItem(itemID, player.getItemAssistant().getItemSlot(itemID), 1);
-					} else {
-						player.getItemAssistant().deleteItem(itemID, fromSlot, 1);
-					}
-					player.getItemAssistant().addItem(995, TotPrice2);
-					addShopItem(itemID, 1);
-					if (player.getPlayerAssistant().isPlayer()) {
-						GameLogger.writeLog(player.playerName, "shopselling", player.playerName + " sold " + itemName + " to store id: " + player.myShopId + " for" + GameLogger.formatCurrency(TotPrice2) + " coins");
-					}
+			int TotPrice2 = (int) Math.floor(getItemShopValue(itemID, amount, true) * amount); //Something about total price of item?
+			if (player.getItemAssistant().freeSlots() > 0 || player.getItemAssistant().playerHasItem(995)) { //Checks to see if player has room for coins.
+				if (!ItemDefinitions.getDef()[itemID].isNoteable) { //Check to see if its notable.
+					player.getItemAssistant().deleteItem(itemID, player.getItemAssistant().getItemSlot(itemID), amount); //don't really understand if the item isn't notable why it still needs to find the slot.
 				} else {
-					player.getActionSender().sendMessage("You don't have enough space in your inventory.");
-					break;
+					player.getItemAssistant().deleteItem(itemID, fromSlot, amount);
+				}
+				player.getItemAssistant().addItem(995, TotPrice2); //Add the coins to your inventory.
+				addShopItem(itemID, amount); //Add item to the shop.
+				if (player.getPlayerAssistant().isPlayer()) { //Logger
+					GameLogger.writeLog(player.playerName, "shopselling", player.playerName + " sold " + itemName + " to store id: " + player.myShopId + " for" + GameLogger.formatCurrency(TotPrice2) + " coins");
+					player.getActionSender().sendMessage("You sold " + amount + " " +itemName + " for " + TotPrice2 + " gp." );
 				}
+			} else {
+				player.getActionSender().sendMessage("You don't have enough space in your inventory.");
 			}
 			player.getItemAssistant().resetItems(3823);
 			resetShop(player.myShopId);
 			updatePlayerShop();
+			player.getActionSender().sendMessage("You sold " + amount + " " +itemName + " for " + TotPrice2 + " gp." );

```

Comment:
It looks like this message is being sent twice?
Already sent on [line#384](https://github.com/dginovker/2006rebotted/pull/61/files#diff-21cdeee7633cd5195bd70dd70768ed7fR384)

Reply:
hmm..

it only shows up once, thats why i missed it probably